### PR TITLE
Added button to follow/unfollow channel

### LINKED ISSUE #1663: Follow / Unfollow button
As a user, I'd like to be able to follow (subscribe to) a .channel, so that it will appear on my front page, in my left nav and I will get notifications of new posts. 

Similarly, I'd like to be able to unfollow a channel. 

As a moderator or a contributor, I'd like to be able to unfollow a channel and stop receiving notifications, even if I retain my moderator / contributor status. 

### acceptance criteria
- [ ] when not subscribed, display "follow" button: https://zpl.io/VQEJqKk
- [ ] when subscribed, display a "following" button with a drop down: https://zpl.io/a3K9o9Q
  - [ ] "Unfollow channel" => remove subscription
  - [ ] "Leave channel" => remove contributor status & remove subscription
- [ ] update user status in reddit and in discussions app

### PR DESCRIPTION
**Pull request by gsidebo**
#### Pre-Flight checklist

- [x] Screenshots and design review for any changes that affect layout or styling
  - [x] Desktop screenshots 
  - [x] Mobile width screenshots 
  - [x] tag @ferdi or @pdpinch for review  
- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested

#### What are the relevant tickets?
Closes #1663

#### What's this PR do?
Adds button to follow channel and controls to unfollow or leave channel to the channel header

#### How should this be manually tested?
Try the follow button and both "following" menu options, and test the UI at different screen widths

#### Any background context you want to provide?
The [design](https://app.zeplin.io/project/5b181d3bf71408ae4d406b94/screen/5bd9df8815d2791b648ee414) shows a "Following" button that looks like it has white text with a transparent background, but I was concerned that depending on the channel header image it wouldn't be visible. I kept the same styling as the "Follow" button [here](https://app.zeplin.io/project/5b181d3bf71408ae4d406b94/screen/5ba4e7e0a4dc98a7608ee6b5)

#### Screenshots (if appropriate)
ss 2019-01-22 at 14 30 23
ss 2019-01-22 at 14 30 15
ss 2019-01-22 at 14 30 08
ss 2019-01-22 at 14 29 53
ss 2019-01-22 at 14 29 42
ss 2019-01-22 at 14 29 34

### CODE REVIEW

**Comment by gsidebo** [comment]
@pdpinch Please review screenshots

File: static/js/containers/ChannelFollowControls.js [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+  isDropdownOpen: boolean,
+  showDropdown: () => void,
+  hideDropdown: () => void,
+  followChannel: () => void,
+  unfollowChannel: () => void,
+  leaveChannel: () => void,
+  loadChannel: () => void
+}
+
+export const DropDownArrow = () => (
```
Comment (rhysyngsun) [inline static/js/containers/ChannelFollowControls.js]: We have an identical definition in `UserMenu`, could you move these into a utility component file instead?

File: static/js/containers/ChannelFollowControls.js [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+        </a>
+        {isDropdownOpen ? (
+          <DropdownMenu
+            closeMenu={hideDropdown}
+            className="channel-follow-dropdown"
+          >
+            <li>
+              <a onClick={this.handleUnfollowClick}>Unfollow channel</a>
+            </li>
+            <li>
```
[the remark this answers is not in the document]
Comment (rhysyngsun) [reply static/js/containers/ChannelFollowControls.js]: A few other cases we shouldn't show the "Leave channel" option:

- Public channels
- Channels I'm not a contributor of

File: static/js/containers/ChannelFollowControls.js [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+    const { leaveChannel, loadChannel } = this.props
+
+    await leaveChannel()
+    await loadChannel()
+  }
+
+  render() {
+    const { channel, showDropdown, hideDropdown, isDropdownOpen } = this.props
+
+    return channel.user_is_subscriber ? (
```
Comment (rhysyngsun) [inline static/js/containers/ChannelFollowControls.js]: It's possible for the user to be a contributor but not a subscriber, which means this logic wouldn't allow them to leave the channel without re-subscribing first.

**Comment by rhysyngsun** [comment]
Also a flow error is failing travis.

File: static/js/containers/ChannelFollowControls.js (hunk 3 above)
Comment (gsidebo) [reply static/js/containers/ChannelFollowControls.js]: I'm going to call this one out of scope. The designs as they are don't call for a "leave channel" option for an unsubscribed user. @pdpinch should we think about a new issue for this?

File: static/js/containers/ChannelFollowControls.js (hunk 1 above)
Comment (gsidebo) [reply static/js/containers/ChannelFollowControls.js]: Good catch. Meant to do this earlier

File: static/js/containers/ChannelFollowControls.js [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+        </a>
+        {isDropdownOpen ? (
+          <DropdownMenu
+            closeMenu={hideDropdown}
+            className="channel-follow-dropdown"
+          >
+            <li>
+              <a onClick={this.handleUnfollowClick}>Unfollow channel</a>
+            </li>
+            {channel.channel_type === CHANNEL_TYPE_PRIVATE &&
```
Comment (rhysyngsun) [inline static/js/containers/ChannelFollowControls.js]: I think this option should be available for restricted channels too?

**Comment by Ferdi** [comment]
@gsidebo  Can you summarize what does it do . I have seen different discussions about this.

**Comment by gsidebo** [comment]
@Ferdi The "follow" button and the "unfollow channel" buttons add and remove the user as a subscriber (respectively). The "leave channel" button (1) removes the user as a subscriber, and (2) removes the user as a contributor (this was discussed with Peter).
